source/CropView.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022-2023. All rights reserved.
+ * Distributed under the terms of the MIT License.
+ *
+ * Andi Machovec (BlueSky), andi.machovec@gmail.com, 2023
+*/
+
+
+#include "CropView.h"
+
+#include <File.h>
+#include <Bitmap.h>
+#include <TranslationUtils.h>
+#include <TranslatorFormats.h>
+
+
+CropView::CropView()
+	:
+	BView("", B_SUPPORTS_LAYOUT|B_WILL_DRAW)
+{
+	fImageLoaded = false;
+	fTopCrop = 0;
+	fBottomCrop = 0;
+	fLeftCrop = 0;
+	fRightCrop = 0;
+}
+
+
+status_t
+CropView::LoadImage(const BString& filename)
+{
+	fBitmap = BTranslationUtils::GetBitmap(filename.String());
+
+	if (fBitmap != nullptr)
+	{
+		if (fBitmap->IsValid())
+		{
+			fImageLoaded = true;
+			fImageSize = fBitmap->Bounds().Size();
+
+			_SetDrawingRect();
+			_SetMarkerRect();
+			Invalidate();
+			return B_OK;
+		}
+	}
+
+	return B_ERROR;
+}
+
+
+void
+CropView::Draw(BRect updateRect)
+{
+	if (fImageLoaded)
+	{
+		DrawBitmap(fBitmap, fBitmap->Bounds(), fDrawingRect);
+
+		if ((fTopCrop+fBottomCrop+fLeftCrop+fRightCrop) > 0) // only draw crop marker when
+		{													 // at least on cropping value is set
+			SetHighColor(255,0,0);
+			StrokeRect(fMarkerRect, B_SOLID_HIGH);
```

Comment:
Thanks for the detailed review @humdingerb . 

> * how about renaming the tabs: Main options -> Options, Cropping options -> Cropping.

Sounds good, I'll change it. 



> * f we're not encoding video (either disabled or 1:1 copy) the whole Cropping tab incl. label should be disabled.

I was unsure if I should do this. But it makes sense. 

> * As there's no "Enable cropping" checkbox any more, maybe a button "Reset" could zero all spinners.

I removed the checkbox because I thought it wasn't really necessary. If all cropping values are at zero  -> no cropping. But a reset button sounds like a good idea. 

> * As the cropping spinner text boxes are a bit wide, and we can easily make the window wider, but not higher, how about moving the crop spinners to the left, and the preview to the right of it? More space for the video, esp. if its in portrait mode (damn you cellphone videos and tiktok! :) )

I agree. I was experimenting with such a layout before and didn't really get it to work like I wanted. Can I leave this one to you for a subsequent PR? ;-)



> * Many movies start with a black frame. Not ideal for cropping. Idea: As we got the video duration, would it be possible to grab 10 evenly spaced frames from the video and switch between them with "<" and ">" buttons?

Yep, this is definitely needed. I'll add it to this PR before it gets merged. 

> * Loading a video and changing to the Cropping tab pegs a CPU.

Didn't notice that -> will investigate. 

> * Last but least: coding style the curly { comes at the end of the "if" line. :)

Is this really defined as such? In the style guide both variants appear. I'll change it of course if needed but personally I hate if the opening curly bracket is not on the same indent as the closing one. Makes it much less clear to me especially if you have several nested blocks.

Reply:
All good, if there's anything you want to postpone to a later PR, just give the word.

WRT the curly guide, at least for if/for/while conditions, the examples (and all code in the Haiku tree) say so: https://www.haiku-os.org/development/coding-guidelines